.github/actions/copy-workflow-go/action.yml

```diff
@@ -28,10 +28,12 @@ runs:
           if [[ $CURRENT_VERSION < $TARGET_VERSION ]]; then
             echo "GO_VERSION_BUMP=1" >> $GITHUB_ENV
 
-            # Update the version in go.mod. This alone ensures there's a diff.
-            go mod edit -go $TARGET_VERSION
-
             # We tidy immediately because some updates require it
+            # We ensure dependencies are resolved to versions supported by CURRENT_VERSION of Go
+            # It also modifies Go version in the go.mod which ensures there's a diff
+            go mod tidy -go=$CURRENT_VERSION && go mod tidy -go=$TARGET_VERSION
```

Comment:
What happens if $CURRENT_VERSION is less than $TARGET_VERSION-1, e.g. because we're updating a repo that hasn't been updated in a long time, or is just getting added to Unified CI.

Reply:
You're right! We want to use `TARGET_VERSION-1` here. I added `PREVIOUS_TARGET_VERSION` variable so that we're explicit about it.